### PR TITLE
Syngas Documentation

### PR DESCRIPTION
**Pull request by AlbertLee125**
I have documented the Syngas case and added the original references of the given problem.

### CODE REVIEW

File: gdplib/syngas/syngas_adapted.py [hunk 1]
```diff
@@ -12,43 +14,65 @@
 
 
 def build_model():
+    """_summary_
```
Comment (bernalde) [inline gdplib/syngas/syngas_adapted.py]: Missing summary

File: gdplib/syngas/syngas_adapted.py [hunk 2]
```diff
@@ -340,330 +841,1063 @@ def unit_exists_or_not(disj, option):
...
-                 + m.p2[equip] * m.syngas_tech_num_units[equip, tech] * m.module_factors[tech, equip])
-                * m.cost_index_ratio / 8000
+                (
+                    m.p1[equip] * m.variable_utilization[tech, equip]
+                    + m.p2[equip]
+                    * m.syngas_tech_num_units[equip, tech]
+                    * m.module_factors[tech, equip]
+                )
+                * m.cost_index_ratio
+                / 8000
```
Comment (bernalde) [inline gdplib/syngas/syngas_adapted.py]: Add a comment that this referes to 8000 hr/ year

File: gdplib/syngas/syngas_adapted.py [hunk 3]
```diff
@@ -340,330 +841,1063 @@ def unit_exists_or_not(disj, option):
...
+            (m.p1['flash'] * 100 * m.flash_water + m.p2['flash'])
+            * m.aux_module_factors['flash']
+            / 8000
+            * m.cost_index_ratio
+        ),
+        doc="the capital cost for the flash unit based on the amount of water separated [$·h-1]",
+    )
 
-    # PSA
+    # PSA(pressure swing adsorption)
```
Comment (bernalde) [inline gdplib/syngas/syngas_adapted.py]: Add space before (

File: gdplib/syngas/syngas_adapted.py [hunk 4]
```diff
@@ -340,330 +841,1063 @@ def unit_exists_or_not(disj, option):
...
+        ----------
+        m : Pyomo.ConcreteModel
+            The concrete model instance which contains sets, parameters, and variables that define the syngas production process.
+        species : str
+            The index of chemical species which is the element of the set 'm.species'.
+
+        Returns
+        -------
+        Pyomo.Constraint
+            Ensures the sum of outflow and purge equals the inflow for the mixer(ms4) across all species.
```
Comment (bernalde) [inline gdplib/syngas/syngas_adapted.py]: Add space before (

File: gdplib/syngas/syngas_adapted.py [hunk 5]
```diff
@@ -340,330 +841,1063 @@ def unit_exists_or_not(disj, option):
...
+        """
+        return (
+            m.flow_out_from['ms4', species] + m.purge_flow[species]
+            == m.flow_into['ms4', species]
+        )
 
     @m.Constraint(m.species)
     def purge_flow_limit(m, species):
+        """
+        Limits the purge flow of each species to no more than 1% of its inflow to the mixer(ms4).
```
Comment (bernalde) [inline gdplib/syngas/syngas_adapted.py]: Add space before (

File: gdplib/syngas/syngas_adapted.py [hunk 6]
```diff
@@ -340,330 +841,1063 @@ def unit_exists_or_not(disj, option):
...
+        ----------
+        m : Pyomo.ConcreteModel
+            The concrete model instance which contains sets, parameters, and variables that define the syngas production process.
+        species : str
+            The index of chemical species which is the element of the set 'm.species'.
+
+        Returns
+        -------
+        Pyomo.Constraint
+            Constrains the purge flow to a maximum of 1% of the respective species' inflow to the MS4, aiding in maintaining environmental and process control standards.
```
Comment (bernalde) [inline gdplib/syngas/syngas_adapted.py]: Are we using upper case (MS4) or lower case (ms4) for units? Standardize it

**Review by bernalde — CHANGES_REQUESTED** [review]
There are a number of missing spaces before parenthesis and some comments to be addressed.

File: gdplib/syngas/syngas_adapted.py (hunk 3 above)
Comment (AlbertLee125) [reply gdplib/syngas/syngas_adapted.py]: space added.

File: gdplib/syngas/syngas_adapted.py (hunk 4 above)
Comment (AlbertLee125) [reply gdplib/syngas/syngas_adapted.py]: Separated the space. And the ms and m both stands for the mixer. However, ms is a mixer/splitter unit and m is mixer itself. I will check the documentation

File: gdplib/syngas/syngas_adapted.py (hunk 6 above)
Comment (AlbertLee125) [reply gdplib/syngas/syngas_adapted.py]: I will unify as lower case.

File: gdplib/syngas/syngas_adapted.py (hunk 2 above)
Comment (AlbertLee125) [reply gdplib/syngas/syngas_adapted.py]: Added # [8000 hr/yr] for all existing 8 cases.

File: gdplib/syngas/syngas_adapted.py (hunk 1 above)
Comment (AlbertLee125) [reply gdplib/syngas/syngas_adapted.py]: Added the documentation. Used the GenAI of Grammarly to fix the grammar error and enhance the Notes footage.

**Review by ZedongPeng — APPROVED** [review]

File: gdplib/syngas/syngas_adapted.py [hunk 7]
```diff
@@ -1533,7 +1538,7 @@ def ms4_outlet_mass_balance(m, species):
     @m.Constraint(m.species)
     def purge_flow_limit(m, species):
         """
-        Limits the purge flow of each species to no more than 1% of its inflow to the mixer(ms4).
+        Limits the purge flow of each species to no more than 1% of its inflow to the mixer\splitter (ms4).
```
Comment (bernalde) [inline gdplib/syngas/syngas_adapted.py]: Wrong slash "/"

File: gdplib/syngas/syngas_adapted.py [hunk 8]
```diff
@@ -1615,7 +1620,7 @@ def s2_mass_balance(m, species):
         Returns
         -------
         Pyomo.Constraint
-            Enforces that the inflow of each species into splitter(s2) equals the outflow from splitter(s2) to the mixer(ms1).
+            Enforces that the inflow of each species into splitter (s2) equals the outflow from splitter (s2) to the mixer\splitter (ms1).
```
Comment (bernalde) [inline gdplib/syngas/syngas_adapted.py]: Wrong slash

**Review by bernalde — APPROVED** [review]
Consider this approved from my side once the slashes are fixed.
Merging contingent on @ZedongPeng and @tristantc reviews

File: gdplib/syngas/syngas_adapted.py (hunk 8 above)
Comment (AlbertLee125) [reply gdplib/syngas/syngas_adapted.py]: slash fixed.

File: gdplib/syngas/syngas_adapted.py (hunk 7 above)
Comment (AlbertLee125) [reply gdplib/syngas/syngas_adapted.py]: slash fixed

**Pull request merged**